lib/p2p/NodeList.ts

```diff
@@ -87,6 +111,43 @@ class NodeList {
     return false;
   }
 
+  /**
+   * Add a reputation event to the node's history
+   * @return true if the specified node exists and the event was added, false otherwise
+   */
+  public addReputationEvent = async (nodePubKey: string, event: ReputationEvent): Promise<boolean> => {
+    const node = this.nodes.get(nodePubKey);
+    if (node) {
+      const promises: PromiseLike<any>[] = [
+        this.repository.addReputationEvent({ event, nodeId: node.id }),
+      ];
+
+      switch (event) {
+        case (ReputationEvent.ManualBan):
+        case (ReputationEvent.ManualUnban): {
+          node.reputationScore = reputationEventWeight[event];
+          break;
+        }
+        default: node.reputationScore += reputationEventWeight[event];
+      }
+
+      if (this.nodeIsBanned(node)) {
+        promises.push(this.setBanStatus(node, true));
+        this.emit('node.ban', nodePubKey);
+      } else if (node.banned) {
+        // If the reputationScore is not below the banThreshold but node.banned
+        // is true that means that the node was unbanned
+        promises.push(this.setBanStatus(node, false));
```

Comment:
Does it make sense to also set `node.banned` to false here?

Reply:
`this.setBanStatus` is doing that and also saving that change to the database